### PR TITLE
chore(demo-playwright): tests for `addon-table`

### PR DESCRIPTION
**Pull request by JoelLau**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [x] Tests related changes
- [ ] Documentation content changes

## What is the current behaviour?

#4912

Existing tests are written in cypress

## What is the new behaviour?

New tests are written in playwright

### CODE REVIEW

**Comment by JoelLau** [comment]
i've attempted to follow the previous test structure as closely as possible - lmk if there are improvements to be made and if this PR should be split by file

File: projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            await tuiGoto(page, `/components/table-pagination/API`);
+            const {apiPageExample} = new TuiDocumentationPagePO(page);
+
+            const {tablePagination, linesPerPageSelect} =
+                await initializeAliases(apiPageExample);
+
+            await linesPerPageSelect.click();
+
+            await waitForCheckmarkIcon(page);
+            await expect(tablePagination).toHaveScreenshot(`0-[size]-dropdown-base.png`);
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts:16]: Let's decrease screenshot area a little bit
0--size--dropdown-base-chromium-darwin

File: projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                `/components/table-pagination/API?items=[0, ${longNumber}]&size=${longNumber}&total=${longNumber}&page=0`,
+            );
+            const {apiPageExample} = new TuiDocumentationPagePO(page);
+            const {tablePagination, linesPerPageSelect} =
+                await initializeAliases(apiPageExample);
+
+            await linesPerPageSelect.click();
+
+            await waitForCheckmarkIcon(page);
+            await expect(tablePagination).toHaveScreenshot(`1-[size]-dropdown-base.png`);
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts:33]: 1--size--dropdown-base-chromium-darwin

File: projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import {expect, Locator, Page, test} from '@playwright/test';
+
+test.describe(`TablePagination`, () => {
+    test.describe(`Dropdown with \`[size]\`-options (amount items per page)`, () => {
+        test(`Basic case`, async ({page}) => {
+            await tuiGoto(page, `/components/table-pagination/API`);
+            const {apiPageExample} = new TuiDocumentationPagePO(page);
+
+            const {tablePagination, linesPerPageSelect} =
+                await initializeAliases(apiPageExample);
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts:11]: Look into other playwright tests.
They use Page object pattern.
Let's create `TuiTablePaginationPO`:

```ts
const tablePagination = new TuiTablePaginationPO(apiPageExample.locator(`tui-table-pagination`));

// [...]

await tablePagination.linesPerPageSelect.click()
```

File: projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+            await linesPerPageSelect.click();
+
+            await waitForCheckmarkIcon(page);
+            await expect(tablePagination).toHaveScreenshot(`1-[size]-dropdown-base.png`);
+        });
+    });
+
+    test(`Custom size-option content`, async ({page}) => {
+        await tuiGoto(page, `/components/table-pagination/API`);
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts:38]: Is it a mistake ? 🤔 

https://github.com/taiga-family/taiga-ui/blob/5d44c31b1fa483a0aa4ebbeb930c8669c0eca917/projects/demo-cypress/cypress/tests/addon-table/table-pagination/table-pagination.cy.ts#L37-L40

https://taiga-ui.dev/components/table-pagination#custom-size-option-content

File: projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            `[automation-id=tui-table-pagination__lines-per-page-wrapper] tui-hosted-dropdown`,
+        );
+
+        await expect(tablePagination).toBeVisible();
+        await expect(linesPerPageSelect).toBeVisible();
+
+        return {tablePagination, linesPerPageSelect};
+    }
+
+    async function waitForCheckmarkIcon(page: Page): Promise<void> {
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts:64]: Let' create `waitForDropdown` inside `TuiTablePaginationPO` and put these code inside this method.

File: projects/demo-playwright/tests/addon-table/table/table.spec.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import {TuiDocumentationPagePO, tuiGoto} from '@demo-playwright/utils';
+import {expect, test} from '@playwright/test';
+
+test.describe(`Table`, () => {
+    test(`Dynamic Column`, async ({page}) => {
+        await tuiGoto(page, `components/table`);
+        const example = new TuiDocumentationPagePO(page).getExample(`#dynamic`);
+        const copyButton = example.locator(`button`).first();
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/addon-table/table/table.spec.ts:8]: Why `copyButton` ?
Should it be `addColumnButton` ?

File: projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts (hunk 2 above)
Comment (nsbarsukov) [reply projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts:33]: (i think we can change width a little bit too)

File: projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts (hunk 1 above)
Comment (JoelLau) [reply projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts:16]: how small should i go?

0--size--dropdown-base-chromium-darwin

File: projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts (hunk 4 above)
Comment (JoelLau) [reply projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts]: yes, i've updated the test to check against the below. my mistake 😓 

2--sizeOptionContent--dropdown

File: projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts (hunk 5 above)
Comment (JoelLau) [reply projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts]: fixed in 1a7f7e6d3d63c38ad73aa828bbd637e8afb0dcb4

File: projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts (hunk 3 above)
Comment (JoelLau) [reply projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts]: fixed in 1a7f7e6d3d63c38ad73aa828bbd637e8afb0dcb4

File: projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts (hunk 1 above)
Comment (JoelLau) [reply projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts:25]: decreased in 7edb545c0c79854fe9b037e5fd74e91c144444f8

File: projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts (hunk 2 above)
Comment (JoelLau) [reply projects/demo-playwright/tests/addon-table/table-pagination/table-pagination.spec.ts:46]: changed in 7edb545c0c79854fe9b037e5fd74e91c144444f8

**Comment by splincode** [comment]
@JoelLau conflicts

**Comment by JoelLau** [comment]
> Fall with an error if diff-output exists

need a bit of help understanding the messages on CI - am i missing some screenshot?

**Review by splincode — APPROVED** [review]

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**